libs/auth/react/src/lib/userManager.ts

```diff
@@ -44,6 +44,7 @@ export const configureMock = (user?: MockUser) => {
     profile: { name: 'Mock', locale: 'is', nationalId: '0000000000' },
     expired: false,
     expires_in: 9999,
+    scopes: [],
```

Comment:
The larger problems here are all these type assertions that totally mask the possibility of all sorts of other problems.

https://github.com/island-is/island.is/blob/ee4c0520fb05c178e93c98075013c8498474276f/libs/auth/react/src/lib/userManager.ts#L43-L68

`userInfo` should be made to be a `User` straight away, and the issue with the profile property should be fixed.

That then obviates the two `as User` assertions further down.

Finally, the trailing `as unknown) as UserManager` is just super smelly, and ought to go.

Reply:
At the very(!) least we should push any icky type-assertions as far down the property tree as possible, to make it more explicit what were lying about.

Reply:
That's a valid point. 
My plan here was to just prevent the error when using the service-portal as a mock user.
Those other changes might be revisited by the code owners at a later time.

Reply:
Makes sense.
Your changes LGTM!

Reply:
UserManager is a huge interface, but it's mostly wrapped (and unit tested) inside this library.

Elsewhere I've started adding our User types to express `request.user` and oicd-client `user.profile`. Hopefully we can apply that everywhere.

Maybe it would be better to completely wrap (hide) oidc-client, but it has so many features that I didn't dare yet.